src/node_options.cc

```diff
@@ -166,9 +166,10 @@ void EnvironmentOptions::CheckOptions(std::vector<std::string>* errors,
           "--watch-path cannot be used in combination with --test");
     }
 
-#ifndef ALLOW_ATTACHING_DEBUGGER_IN_TEST_RUNNER
-    debug_options_.allow_attaching_debugger = false;
-#endif
+    if (watch_mode && test_isolation == "none") {
+      errors->push_back(
+          "--watch cannot be used with --experimental-test-isolation=none");
+    }
```

Comment:
Why we can't use watch with no isolation?

We should update the docs to reflect that if we keep this

Reply:
since `--watch` always requires a parent process to restart the internal one when needed